src/recipe.py

```diff
@@ -0,0 +1,41 @@
+# pragma: no cover
+from src.config import DuneToLocalJob, Env, LocalToDuneJob
+from src.destinations.dune import DuneDestination
+from src.destinations.postgres import PostgresDestination
+from src.logger import log
+from src.sources.dune import DuneSource
+from src.sources.postgres import PostgresSource
+
+
```

Comment:
These two methods are only temporary while we still have these DuneToLocal and LocalToDune classes. 

Once we have a single Job class. it will be something like:

```
@dataclass
class Job(Generic[T]):
    source: Source[T]
    destination: Destination
    env: Env

    def validate(self) -> None:
        if not self.source.validate():
            raise ValueError(f"Invalid {self.source.__class__.__name__} configuration")
        if not self.destination.validate():
            raise ValueError(f"Invalid {self.destination.__class__.__name__} configuration")

    def run(self) -> None:
        """Execute the job by fetching from source and saving to destination"""
        self.validate()
        data = self.source.fetch()
        
        # Handle the specific case for DuneSource which returns Tuple[DataFrame, str]
        if isinstance(data, tuple) and isinstance(data[0], DataFrame):
            if not data[0].empty:
                self.destination.save(data)
            else:
                log.warning("No Query results found! Skipping write")
        else:
            self.destination.save(data)
```